scripts/diff-workflow-against-master.sh

```diff
@@ -0,0 +1,471 @@
+#!/bin/bash
+
+# Copyright (C) 2020  Kevin R Croft <krcroft@gmail.com>
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+##
+#  This script craws the current repo's GitHub workflow content.
+#  It fetches assets and logs from the most recent successful master
+#  run followed by the most recent (and possibly failing) current
+#  branch, which can also be a master branch.
+#  The goal of this script is two fold:
+#    - Provide a mechanized an automated way to fetch CI records.
+#    - Provide a rapid way to diff bad CI runs against master.
+#
+#  This script requires a GitHub account in order to generate an
+#  auth-token. Simply run the script, it will provide instructions.
+#
+set -euo pipefail
+shopt -s nullglob
+
+# Fixed portion of the URL
+declare -gr scheme="https://"
+declare -gr authority="api.github.com"
+
+# Colors
+declare -gr bold="\\e[1m"
+declare -gr red="\\e[31m"
+declare -gr green="\\e[32m"
+declare -gr yellow="\\e[33m"
+declare -gr cyan="\\e[36m"
+declare -gr reset="\\e[0m"
+
+##
+#  Changes the working directory to that of the
+#  repository's root.
+#
+function cd_repo_root() {
+	if [[ "${in_root:-}" != "true" ]]; then
+		script_path="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+		pushd "$script_path" > /dev/null
+		pushd "$(git rev-parse --show-toplevel)" > /dev/null
+		in_root=true
+	fi
+}
+
+##
+#  Determines the full GitHub repo name using the
+#  remote origin set in the repository's configuration
+#
+function init_baseurl() {
+	cd_repo_root
+	# Extract the full GitHub repo name from the origin
+	repo="$(git config --get remote.origin.url | sed 's/.*://;s/\.git$//;s^//[^/]*/^^')"
+	baseurl="$scheme/$authority/repos/$repo/actions"
+	declare -gr repo
+	declare -gr baseurl
+}
+
+##
+#  Determines the local branch name
+#
+function init_local_branch() {
+	cd_repo_root
+	local_branch="$(git rev-parse --abbrev-ref HEAD)"
+	declare -gr local_branch
+}
+
+##
+#  Sets up NETRC credentials for GitHub's v3 API server.
+#  NETRC is currently the most secure way to provide
+#  credentials to CURL, because it prevents other processes
+#  from inspect the process environment and cmd arguments;
+#  both of which can be found in /proc).
+#
+function init_netrc() {
+	# Check and setup the GitHub personal access token
+	local netrc="$HOME/.netrc"
+	if grep -q "^machine $authority" "$netrc"; then
+		echo "Found credentials for $authority in $netrc"
+		return
+	fi
+	# Get the username
+	clear
```

Comment:
Don't clear screen, please.

Reply:
It felt a bit more instructive, but I'm 50-50 on how it turned out. A real  TUI would have been cool with something like `dialog` :-)  But yeah, will drop the `clear`s!

Reply:
Done!